Build fails on i386: error: conversion function cannot be redeclared

```
In file included from /wrkdirs/usr/ports/cad/surelog/work/Surelog-1.36/src/Library/ParseLibraryDef.cpp:25:
In file included from /wrkdirs/usr/ports/cad/surelog/work/Surelog-1.36/include/Surelog/Config/ConfigSet.h:28:
In file included from /wrkdirs/usr/ports/cad/surelog/work/Surelog-1.36/include/Surelog/Config/Config.h:29:
/wrkdirs/usr/ports/cad/surelog/work/Surelog-1.36/include/Surelog/Common/NodeId.h:27:3: error: conversion function cannot be redeclared
  operator std::size_t() const { return id; }
  ^
/wrkdirs/usr/ports/cad/surelog/work/Surelog-1.36/include/Surelog/Common/NodeId.h:26:3: note: previous definition is here
  operator RawNodeId() const { return id; }
  ^
1 error generated.
```

Version: 1.36
clang-14
FreeBSD 13.1

--- a/include/Surelog/Common/NodeId.h
+++ b/include/Surelog/Common/NodeId.h
@@ -6,6 +6,7 @@
 #include <ostream>
 #include <set>
 #include <unordered_set>
+#include <type_traits>
 
 namespace SURELOG {
 /**
@@ -24,7 +25,15 @@ class NodeId final {
   constexpr NodeId(const NodeId &rhs) : id(rhs.id) {}
 
   operator RawNodeId() const { return id; }
-  operator std::size_t() const { return id; }
+
+  // Don't include size_t conversion on 32Bit machines where sizeof(size_t)==32
+  template <typename T = std::size_t,
+            typename std::enable_if<!std::is_same<T, RawNodeId>::value>::type
+                * = nullptr>
+  operator std::size_t() const {
+    return id;
+  }
+
   operator bool() const { return id != InvalidRawNodeId; }
 
   bool operator<(const NodeId &rhs) const { return id < rhs.id; }

